qiskit/circuit/library/hidden_linear_function.py

```diff
@@ -80,6 +81,10 @@ def __init__(self,
             adjacency_matrix: a symmetric n-by-n list of 0-1 lists.
                 n will be the number of qubits.
         """
+        adjacency_matrix = np.array(adjacency_matrix)
```

Comment:
Using `asarray` avoids copying the matrix.
```suggestion
        adjacency_matrix = np.asarray(adjacency_matrix)
```